com.hazelcast.jet.job.JobSubmissionSlownessRegressionTest.regressionTestForPR1488: java.lang.OutOfMemoryError: Java heap space

_master_ (commit ddb94a38d54476d7a4cb65258e63fba5158652f2)

Failed on nightly build: http://jenkins.hazelcast.com/job/jet-oss-master-nightly/64/testReport/junit/com.hazelcast.jet.job/JobSubmissionSlownessRegressionTest/regressionTestForPR1488/

Standard Error output:
```
Exception in thread "hz._hzInstance_1_jet.SlowOperationDetectorThread" java.lang.OutOfMemoryError: Java heap space
THREAD DUMP FOR TEST FAILURE: "test timed out after 600000 milliseconds" at "regressionTestForPR1488"

	at java.util.Arrays.copyOf(Arrays.java:3332)
	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:448)
	at java.lang.StringBuilder.append(StringBuilder.java:136)
	at java.lang.StackTraceElement.toString(StackTraceElement.java:173)
	at com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector$DetectorThread.getStackTraceOrNull(SlowOperationDetector.java:208)
	at com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector$DetectorThread.scanOperationRunner(SlowOperationDetector.java:194)
	at com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector$DetectorThread.scan(SlowOperationDetector.java:162)
	at com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector$DetectorThread.run(SlowOperationDetector.java:146)
java.lang.OutOfMemoryError: Java heap space
	at com.hazelcast.internal.metrics.impl.MetricsRegistryImpl.incrementMod(MetricsRegistryImpl.java:205)
	at com.hazelcast.internal.metrics.impl.MetricsRegistryImpl.deregister(MetricsRegistryImpl.java:265)
	at com.hazelcast.jet.impl.execution.ExecutionContext$$Lambda$472/539477754.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at com.hazelcast.jet.impl.execution.ExecutionContext.completeExecution(ExecutionContext.java:186)
	at com.hazelcast.jet.impl.JobExecutionService.lambda$completeExecution$10(JobExecutionService.java:320)
	at com.hazelcast.jet.impl.JobExecutionService$$Lambda$469/1744261799.run(Unknown Source)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:626)
	at com.hazelcast.jet.impl.JobExecutionService.completeExecution(JobExecutionService.java:319)
	at com.hazelcast.jet.impl.operation.CompleteExecutionOperation.run(CompleteExecutionOperation.java:79)
	at com.hazelcast.spi.Operation.call(Operation.java:170)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:210)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:407)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:434)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:648)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:633)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:592)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:256)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:61)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:260)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:244)
	at com.hazelcast.jet.impl.MasterJobContext.invokeCompleteExecution(MasterJobContext.java:557)
	at com.hazelcast.jet.impl.MasterJobContext.onExecuteStepCompleted(MasterJobContext.java:472)Unable to get thread dump!

	at com.hazelcast.jet.impl.MasterJobContext$$Lambda$380/2063404290.accept(Unknown Source)
	at com.hazelcast.jet.impl.MasterContext.lambda$invokeOnParticipant$1(MasterContext.java:278)
	at com.hazelcast.jet.impl.MasterContext$$Lambda$372/1838023944.accept(Unknown Source)
	at com.hazelcast.jet.impl.util.Util$2.onResponse(Util.java:173)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:256)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
Exception in thread "surefire-forkedjvm-command-thread" java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
	at java.util.concurrent.ConcurrentHashMap$EntryIterator.next(ConcurrentHashMap.java:3461)
	at java.util.concurrent.ConcurrentHashMap$EntryIterator.next(ConcurrentHashMap.java:3446)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationMonitor$MonitorInvocationsTask.run0(InvocationMonitor.java:318)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationMonitor$FixedRateMonitorTask.run(InvocationMonitor.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.lang.OutOfMemoryError: Java heap space
	at java.util.concurrent.ConcurrentHashMap$EntryIterator.next(ConcurrentHashMap.java:3461)
	at java.util.concurrent.ConcurrentHashMap$EntryIterator.next(ConcurrentHashMap.java:3446)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationMonitor$MonitorInvocationsTask.run0(InvocationMonitor.java:318)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationMonitor$FixedRateMonitorTask.run(InvocationMonitor.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
```

Standard output:
```
Hiccups measured while running test 'regressionTestForPR1488(com.hazelcast.jet.job.JobSubmissionSlownessRegressionTest):'
23:18:45, accumulated pauses: 45 ms, max pause: 33 ms, pauses over 1000 ms: 0
23:18:50, accumulated pauses: 1520 ms, max pause: 329 ms, pauses over 1000 ms: 0
23:18:55, accumulated pauses: 1889 ms, max pause: 168 ms, pauses over 1000 ms: 0
23:19:00, accumulated pauses: 3136 ms, max pause: 519 ms, pauses over 1000 ms: 0
23:19:05, accumulated pauses: 2284 ms, max pause: 775 ms, pauses over 1000 ms: 0
23:19:10, accumulated pauses: 2043 ms, max pause: 221 ms, pauses over 1000 ms: 0
23:19:15, accumulated pauses: 2420 ms, max pause: 475 ms, pauses over 1000 ms: 0
23:19:20, accumulated pauses: 2244 ms, max pause: 558 ms, pauses over 1000 ms: 0
23:19:25, accumulated pauses: 2766 ms, max pause: 537 ms, pauses over 1000 ms: 0
23:19:30, accumulated pauses: 2509 ms, max pause: 447 ms, pauses over 1000 ms: 0
23:19:35, accumulated pauses: 2328 ms, max pause: 393 ms, pauses over 1000 ms: 0
23:19:40, accumulated pauses: 1363 ms, max pause: 244 ms, pauses over 1000 ms: 0
23:19:45, accumulated pauses: 2056 ms, max pause: 469 ms, pauses over 1000 ms: 0
23:19:50, accumulated pauses: 1323 ms, max pause: 151 ms, pauses over 1000 ms: 0
23:19:55, accumulated pauses: 1869 ms, max pause: 379 ms, pauses over 1000 ms: 0
23:20:00, accumulated pauses: 2236 ms, max pause: 325 ms, pauses over 1000 ms: 0
23:20:05, accumulated pauses: 2699 ms, max pause: 524 ms, pauses over 1000 ms: 0
23:20:10, accumulated pauses: 2163 ms, max pause: 619 ms, pauses over 1000 ms: 0
23:20:15, accumulated pauses: 2651 ms, max pause: 816 ms, pauses over 1000 ms: 0
23:20:20, accumulated pauses: 3503 ms, max pause: 1316 ms, pauses over 1000 ms: 1
23:20:25, accumulated pauses: 3424 ms, max pause: 685 ms, pauses over 1000 ms: 0
23:20:30, accumulated pauses: 2632 ms, max pause: 516 ms, pauses over 1000 ms: 0
23:20:35, accumulated pauses: 3963 ms, max pause: 1211 ms, pauses over 1000 ms: 1
23:20:40, accumulated pauses: 3727 ms, max pause: 638 ms, pauses over 1000 ms: 0
23:20:45, accumulated pauses: 3792 ms, max pause: 631 ms, pauses over 1000 ms: 0
23:20:50, accumulated pauses: 3909 ms, max pause: 868 ms, pauses over 1000 ms: 0
23:20:55, accumulated pauses: 4190 ms, max pause: 762 ms, pauses over 1000 ms: 0
23:21:00, accumulated pauses: 3998 ms, max pause: 518 ms, pauses over 1000 ms: 0
23:21:05, accumulated pauses: 4186 ms, max pause: 744 ms, pauses over 1000 ms: 0
23:21:10, accumulated pauses: 4643 ms, max pause: 707 ms, pauses over 1000 ms: 0
23:21:15, accumulated pauses: 3882 ms, max pause: 477 ms, pauses over 1000 ms: 0
23:21:20, accumulated pauses: 4460 ms, max pause: 530 ms, pauses over 1000 ms: 0
23:21:25, accumulated pauses: 4247 ms, max pause: 763 ms, pauses over 1000 ms: 0
23:21:30, accumulated pauses: 4675 ms, max pause: 729 ms, pauses over 1000 ms: 0
23:21:35, accumulated pauses: 5202 ms, max pause: 871 ms, pauses over 1000 ms: 0
23:21:40, accumulated pauses: 3658 ms, max pause: 790 ms, pauses over 1000 ms: 0
23:21:45, accumulated pauses: 4047 ms, max pause: 379 ms, pauses over 1000 ms: 0
23:21:50, accumulated pauses: 4775 ms, max pause: 772 ms, pauses over 1000 ms: 0
23:21:55, accumulated pauses: 4383 ms, max pause: 518 ms, pauses over 1000 ms: 0
23:22:00, accumulated pauses: 4833 ms, max pause: 577 ms, pauses over 1000 ms: 0
23:22:05, accumulated pauses: 5231 ms, max pause: 739 ms, pauses over 1000 ms: 0
23:22:10, accumulated pauses: 4029 ms, max pause: 1152 ms, pauses over 1000 ms: 1
23:22:15, accumulated pauses: 4157 ms, max pause: 576 ms, pauses over 1000 ms: 0
23:22:20, accumulated pauses: 4378 ms, max pause: 307 ms, pauses over 1000 ms: 0
23:22:25, accumulated pauses: 4799 ms, max pause: 616 ms, pauses over 1000 ms: 0
23:22:30, accumulated pauses: 4579 ms, max pause: 496 ms, pauses over 1000 ms: 0
23:22:35, accumulated pauses: 4833 ms, max pause: 652 ms, pauses over 1000 ms: 0
23:22:40, accumulated pauses: 4920 ms, max pause: 1458 ms, pauses over 1000 ms: 1
23:22:45, accumulated pauses: 4590 ms, max pause: 602 ms, pauses over 1000 ms: 0
23:22:50, accumulated pauses: 4842 ms, max pause: 970 ms, pauses over 1000 ms: 0
23:22:55, accumulated pauses: 4926 ms, max pause: 1033 ms, pauses over 1000 ms: 1
23:23:00, accumulated pauses: 4524 ms, max pause: 1381 ms, pauses over 1000 ms: 1
23:23:05, accumulated pauses: 4971 ms, max pause: 943 ms, pauses over 1000 ms: 0
23:23:10, accumulated pauses: 5449 ms, max pause: 2972 ms, pauses over 1000 ms: 2
23:23:15, accumulated pauses: 12896 ms, max pause: 12896 ms, pauses over 1000 ms: 1
23:23:25, accumulated pauses: 5553 ms, max pause: 5553 ms, pauses over 1000 ms: 1
23:23:40, accumulated pauses: 15066 ms, max pause: 15066 ms, pauses over 1000 ms: 1
23:23:45, accumulated pauses: 9834 ms, max pause: 9834 ms, pauses over 1000 ms: 1
23:23:55, accumulated pauses: 1037 ms, max pause: 1037 ms, pauses over 1000 ms: 1
23:24:00, accumulated pauses: 4089 ms, max pause: 4089 ms, pauses over 1000 ms: 1
23:24:05, accumulated pauses: 20944 ms, max pause: 20944 ms, pauses over 1000 ms: 1
23:24:25, accumulated pauses: 7827 ms, max pause: 7827 ms, pauses over 1000 ms: 1
23:24:30, accumulated pauses: 11867 ms, max pause: 11867 ms, pauses over 1000 ms: 1
23:24:45, accumulated pauses: 10242 ms, max pause: 10242 ms, pauses over 1000 ms: 1
23:24:55, accumulated pauses: 2419 ms, max pause: 2419 ms, pauses over 1000 ms: 1
23:25:05, accumulated pauses: 12807 ms, max pause: 12807 ms, pauses over 1000 ms: 1
23:25:10, accumulated pauses: 3675 ms, max pause: 3675 ms, pauses over 1000 ms: 1
23:25:20, accumulated pauses: 12269 ms, max pause: 12269 ms, pauses over 1000 ms: 1
23:25:35, accumulated pauses: 11170 ms, max pause: 10710 ms, pauses over 1000 ms: 1
23:25:40, accumulated pauses: 8267 ms, max pause: 8267 ms, pauses over 1000 ms: 1
23:25:45, accumulated pauses: 7153 ms, max pause: 4350 ms, pauses over 1000 ms: 2
23:25:50, accumulated pauses: 8388 ms, max pause: 8388 ms, pauses over 1000 ms: 1
23:26:00, accumulated pauses: 6236 ms, max pause: 6236 ms, pauses over 1000 ms: 1
23:26:15, accumulated pauses: 15922 ms, max pause: 15922 ms, pauses over 1000 ms: 1
23:26:30, accumulated pauses: 12341 ms, max pause: 12341 ms, pauses over 1000 ms: 1
23:26:35, accumulated pauses: 10225 ms, max pause: 10225 ms, pauses over 1000 ms: 1
23:26:55, accumulated pauses: 23804 ms, max pause: 23804 ms, pauses over 1000 ms: 1
23:27:10, accumulated pauses: 4200 ms, max pause: 4200 ms, pauses over 1000 ms: 1
23:27:15, accumulated pauses: 12790 ms, max pause: 9631 ms, pauses over 1000 ms: 2
23:27:25, accumulated pauses: 7423 ms, max pause: 7423 ms, pauses over 1000 ms: 1
23:27:35, accumulated pauses: 10485 ms, max pause: 10485 ms, pauses over 1000 ms: 1
23:27:45, accumulated pauses: 9260 ms, max pause: 6614 ms, pauses over 1000 ms: 2
23:27:50, accumulated pauses: 9840 ms, max pause: 9840 ms, pauses over 1000 ms: 1
23:28:00, accumulated pauses: 5014 ms, max pause: 5014 ms, pauses over 1000 ms: 1
23:28:05, accumulated pauses: 4627 ms, max pause: 4627 ms, pauses over 1000 ms: 1
23:28:10, accumulated pauses: 6233 ms, max pause: 5593 ms, pauses over 1000 ms: 1
23:28:20, accumulated pauses: 4397 ms, max pause: 4397 ms, pauses over 1000 ms: 1
23:28:30, accumulated pauses: 12024 ms, max pause: 12024 ms, pauses over 1000 ms: 1
23:28:40, accumulated pauses: 12543 ms, max pause: 12543 ms, pauses over 1000 ms: 1
23:28:45, accumulated pauses: 2804 ms, max pause: 2804 ms, pauses over 1000 ms: 1
23:28:50, accumulated pauses: 12387 ms, max pause: 11813 ms, pauses over 1000 ms: 1
23:29:05, accumulated pauses: 7103 ms, max pause: 7103 ms, pauses over 1000 ms: 1
23:29:10, accumulated pauses: 8693 ms, max pause: 6768 ms, pauses over 1000 ms: 2
23:29:20, accumulated pauses: 6385 ms, max pause: 4286 ms, pauses over 1000 ms: 2
23:29:30, accumulated pauses: 9344 ms, max pause: 9344 ms, pauses over 1000 ms: 1
23:29:35, accumulated pauses: 5521 ms, max pause: 5521 ms, pauses over 1000 ms: 1
23:29:40, accumulated pauses: 5709 ms, max pause: 5709 ms, pauses over 1000 ms: 1
23:29:45, accumulated pauses: 2884 ms, max pause: 2884 ms, pauses over 1000 ms: 1
23:29:50, accumulated pauses: 6750 ms, max pause: 6750 ms, pauses over 1000 ms: 1
23:29:55, accumulated pauses: 12132 ms, max pause: 12132 ms, pauses over 1000 ms: 1
23:30:20, accumulated pauses: 19357 ms, max pause: 19357 ms, pauses over 1000 ms: 1
23:30:30, accumulated pauses: 7382 ms, max pause: 7382 ms, pauses over 1000 ms: 1
23:30:40, accumulated pauses: 10943 ms, max pause: 10943 ms, pauses over 1000 ms: 1
23:30:45, accumulated pauses: 7113 ms, max pause: 7113 ms, pauses over 1000 ms: 1
23:30:55, accumulated pauses: 6339 ms, max pause: 6339 ms, pauses over 1000 ms: 1
23:31:00, accumulated pauses: 4519 ms, max pause: 1932 ms, pauses over 1000 ms: 2
23:31:05, accumulated pauses: 5013 ms, max pause: 2724 ms, pauses over 1000 ms: 2
23:31:10, accumulated pauses: 7962 ms, max pause: 7962 ms, pauses over 1000 ms: 1
23:31:15, accumulated pauses: 9345 ms, max pause: 9345 ms, pauses over 1000 ms: 1
23:31:25, accumulated pauses: 13200 ms, max pause: 13200 ms, pauses over 1000 ms: 1
23:31:40, accumulated pauses: 8006 ms, max pause: 8006 ms, pauses over 1000 ms: 1
23:31:50, accumulated pauses: 8557 ms, max pause: 8557 ms, pauses over 1000 ms: 1
23:31:55, accumulated pauses: 11090 ms, max pause: 11090 ms, pauses over 1000 ms: 1
23:32:20, accumulated pauses: 18091 ms, max pause: 12602 ms, pauses over 1000 ms: 3
23:32:25, accumulated pauses: 6183 ms, max pause: 3958 ms, pauses over 1000 ms: 2
23:32:30, accumulated pauses: 4127 ms, max pause: 2741 ms, pauses over 1000 ms: 2
23:32:35, accumulated pauses: 11179 ms, max pause: 9723 ms, pauses over 1000 ms: 2
23:32:45, accumulated pauses: 11333 ms, max pause: 10534 ms, pauses over 1000 ms: 1
23:32:55, accumulated pauses: 13340 ms, max pause: 13340 ms, pauses over 1000 ms: 1
23:33:10, accumulated pauses: 11397 ms, max pause: 11397 ms, pauses over 1000 ms: 1
23:33:20, accumulated pauses: 5288 ms, max pause: 5288 ms, pauses over 1000 ms: 1
23:33:30, accumulated pauses: 5624 ms, max pause: 5624 ms, pauses over 1000 ms: 1
23:33:35, accumulated pauses: 5776 ms, max pause: 1677 ms, pauses over 1000 ms: 2
23:33:40, accumulated pauses: 3803 ms, max pause: 3803 ms, pauses over 1000 ms: 1
23:33:45, accumulated pauses: 6569 ms, max pause: 6569 ms, pauses over 1000 ms: 1
23:33:50, accumulated pauses: 10351 ms, max pause: 10351 ms, pauses over 1000 ms: 1
23:34:00, accumulated pauses: 3213 ms, max pause: 761 ms, pauses over 1000 ms: 0
23:34:05, accumulated pauses: 5225 ms, max pause: 3582 ms, pauses over 1000 ms: 2
23:34:10, accumulated pauses: 2377 ms, max pause: 2377 ms, pauses over 1000 ms: 1
23:34:15, accumulated pauses: 8492 ms, max pause: 8492 ms, pauses over 1000 ms: 1
23:34:20, accumulated pauses: 5124 ms, max pause: 5124 ms, pauses over 1000 ms: 1
23:34:25, accumulated pauses: 5312 ms, max pause: 2799 ms, pauses over 1000 ms: 2
23:34:30, accumulated pauses: 4550 ms, max pause: 4289 ms, pauses over 1000 ms: 1
23:34:35, accumulated pauses: 3680 ms, max pause: 2273 ms, pauses over 1000 ms: 2
23:34:40, accumulated pauses: 8380 ms, max pause: 4124 ms, pauses over 1000 ms: 2
23:34:45, accumulated pauses: 3071 ms, max pause: 3071 ms, pauses over 1000 ms: 1
23:34:50, accumulated pauses: 3657 ms, max pause: 1702 ms, pauses over 1000 ms: 2
23:34:55, accumulated pauses: 5874 ms, max pause: 1781 ms, pauses over 1000 ms: 3
23:35:00, accumulated pauses: 6676 ms, max pause: 6161 ms, pauses over 1000 ms: 1
23:35:05, accumulated pauses: 401 ms, max pause: 401 ms, pauses over 1000 ms: 0
23:35:20, accumulated pauses: 17355 ms, max pause: 15430 ms, pauses over 1000 ms: 1
23:35:25, accumulated pauses: 2305 ms, max pause: 1303 ms, pauses over 1000 ms: 1
23:35:30, accumulated pauses: 12100 ms, max pause: 5391 ms, pauses over 1000 ms: 3
23:35:40, accumulated pauses: 7549 ms, max pause: 3985 ms, pauses over 1000 ms: 2
23:35:45, accumulated pauses: 2789 ms, max pause: 2789 ms, pauses over 1000 ms: 1
23:35:50, accumulated pauses: 5000 ms, max pause: 4310 ms, pauses over 1000 ms: 1
23:35:55, accumulated pauses: 8069 ms, max pause: 8069 ms, pauses over 1000 ms: 1
23:36:00, accumulated pauses: 1616 ms, max pause: 1342 ms, pauses over 1000 ms: 1
23:36:05, accumulated pauses: 4842 ms, max pause: 653 ms, pauses over 1000 ms: 0
23:36:10, accumulated pauses: 6125 ms, max pause: 2421 ms, pauses over 1000 ms: 2
23:36:15, accumulated pauses: 5131 ms, max pause: 4742 ms, pauses over 1000 ms: 1
23:36:20, accumulated pauses: 2870 ms, max pause: 2557 ms, pauses over 1000 ms: 1
23:36:25, accumulated pauses: 6201 ms, max pause: 2493 ms, pauses over 1000 ms: 2
23:36:30, accumulated pauses: 4157 ms, max pause: 1418 ms, pauses over 1000 ms: 2
23:36:35, accumulated pauses: 0 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

Reopened. Unfortunately it happened again: http://jenkins.hazelcast.com/job/jet-oss-master-nightly/71/testReport/junit/com.hazelcast.jet.job/JobSubmissionSlownessRegressionTest/regressionTestForPR1488/

Stacktrace:
```
java.util.concurrent.ExecutionException: java.util.concurrent.RejectedExecutionException: Internal executor rejected task: java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@6e7874c8, because client is shutting down...
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.hazelcast.jet.job.JobSubmissionSlownessRegressionTest.regressionTestForPR1488(JobSubmissionSlownessRegressionTest.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.RejectedExecutionException: Internal executor rejected task: java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@6e7874c8, because client is shutting down...
	at com.hazelcast.client.spi.impl.ClientExecutionServiceImpl$1.rejectedExecution(ClientExecutionServiceImpl.java:75)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
	at java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:622)
	at com.hazelcast.client.spi.impl.ClientExecutionServiceImpl.execute(ClientExecutionServiceImpl.java:107)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.unblock(AbstractInvocationFuture.java:247)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.andThen(AbstractInvocationFuture.java:223)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.andThen(ClientInvocationFuture.java:85)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.andThen(AbstractInvocationFuture.java:213)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.andThen(ClientInvocationFuture.java:80)
	at com.hazelcast.jet.impl.ClientJobProxy$CancellableFuture.andThen(ClientJobProxy.java:227)
	at com.hazelcast.jet.impl.AbstractJobProxy$SubmitJobCallback.resubmitJob(AbstractJobProxy.java:259)
	at com.hazelcast.jet.impl.AbstractJobProxy$SubmitJobCallback.onFailure(AbstractJobProxy.java:249)
	at com.hazelcast.jet.impl.ClientJobProxy$CancellableFuture$1.onFailure(ClientJobProxy.java:235)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onFailure(ClientInvocationFuture.java:139)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onFailure(ClientInvocationFuture.java:139)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:254)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

Standard output is the same:
```
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.util.concurrent.atomic.AtomicReferenceArray.<init>(AtomicReferenceArray.java:94)
	at com.hazelcast.internal.util.concurrent.AbstractConcurrentArrayQueue.<init>(AbstractConcurrentArrayQueue.java:84)
	at com.hazelcast.internal.util.concurrent.OneToOneConcurrentArrayQueue.<init>(OneToOneConcurrentArrayQueue.java:34)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$null$13(ExecutionPlan.java:459)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan$$Lambda$586/1359490148.apply(Unknown Source)
	at java.util.Arrays.setAll(Arrays.java:4698)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$createConveyorArray$14(ExecutionPlan.java:459)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan$$Lambda$585/943633664.apply(Unknown Source)
	at java.util.Arrays.setAll(Arrays.java:4698)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.createConveyorArray(ExecutionPlan.java:457)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$createOutboundCollectors$19(ExecutionPlan.java:518)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan$$Lambda$584/571824338.apply(Unknown Source)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1127)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.createOutboundCollectors(ExecutionPlan.java:515)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.createOutboundEdgeStream(ExecutionPlan.java:470)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.createOutboundEdgeStreams(ExecutionPlan.java:389)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:220)
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:116)
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$9(JobExecutionService.java:214)
	at com.hazelcast.jet.impl.JobExecutionService$$Lambda$561/785001212.run(Unknown Source)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:440)
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:214)
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:72)
	at com.hazelcast.spi.Operation.call(Operation.java:170)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:210)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:407)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:434)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:648)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:633)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:592)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:256)
Exception in thread "pool-3-thread-1" java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.util.jar.Attributes.read(Attributes.java:394)
	at java.util.jar.Manifest.read(Manifest.java:199)
	at java.util.jar.Manifest.<init>(Manifest.java:69)
	at java.util.jar.JarFile.getManifestFromReference(JarFile.java:199)
	at java.util.jar.JarFile.getManifest(JarFile.java:180)
	at sun.misc.URLClassPath$JarLoader$2.getManifest(URLClassPath.java:981)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:450)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.hazelcast.test.mocknetwork.MockNetworkingService$MockEndpointManager.fireConnectionRemovedEvent(MockNetworkingService.java:204)
	at com.hazelcast.test.mocknetwork.MockNetworkingService$MockEndpointManager.access$1200(MockNetworkingService.java:89)
	at com.hazelcast.test.mocknetwork.MockNetworkingService$MockEndpointManager$MockConnLifecycleListener.onConnectionClose(MockNetworkingService.java:293)
	at com.hazelcast.test.mocknetwork.MockNetworkingService$MockEndpointManager$MockConnLifecycleListener.onConnectionClose(MockNetworkingService.java:272)
	at com.hazelcast.test.mocknetwork.MockConnection.close(MockConnection.java:147)
	at com.hazelcast.client.test.TestClientRegistry$MockedNodeConnection.close(TestClientRegistry.java:417)
	at com.hazelcast.client.test.TestClientRegistry$MockedClientConnection$3.run(TestClientRegistry.java:304)
	at com.hazelcast.client.test.TwoWayBlockableExecutor$BlockableRunnable.run(TwoWayBlockableExecutor.java:97)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
THREAD DUMP FOR TEST FAILURE: "java.util.concurrent.RejectedExecutionException: Internal executor rejected task: java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@6e7874c8, because client is shutting down..." at "regressionTestForPR1488"

java.lang.OutOfMemoryError: GC overhead limit exceeded
	at com.hazelcast.internal.partition.impl.PartitionReplicaManager$PartitionAntiEntropyTaskFactory.create(PartitionReplicaManager.java:515)
	at com.hazelcast.internal.partition.impl.PartitionReplicaManager$PartitionAntiEntropyTaskFactory.create(PartitionReplicaManager.java:512)
	at com.hazelcast.spi.impl.operationexecutor.impl.TaskBatch.next(TaskBatch.java:54)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:169)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:131)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110)
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at com.hazelcast.internal.partition.impl.PartitionReplicaManager$PartitionAntiEntropyTaskFactory.create(PartitionReplicaManager.java:515)
	at com.hazelcast.internal.partition.impl.PartitionReplicaManager$PartitionAntiEntropyTaskFactory.create(PartitionReplicaManager.java:512)
	at com.hazelcast.spi.impl.operationexecutor.impl.TaskBatch.next(TaskBatch.java:54)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:169)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:131)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110)
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
java.lang.OutOfMemoryError: GC overhead limit exceeded
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/job/JobSubmissionSlownessRegressionTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/job/JobSubmissionSlownessRegressionTest.java
@@ -17,6 +17,7 @@
 package com.hazelcast.jet.job;
 
 import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.config.EdgeConfig;
 import com.hazelcast.jet.config.JetConfig;
 import com.hazelcast.jet.config.JobConfig;
 import com.hazelcast.jet.core.DAG;
@@ -26,6 +27,10 @@ import com.hazelcast.jet.core.Vertex;
 import com.hazelcast.jet.test.IgnoredForCoverage;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -35,10 +40,6 @@ import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
 
 import static com.hazelcast.jet.core.processor.Processors.noopP;
 import static org.junit.Assert.assertTrue;
@@ -124,7 +125,7 @@ public final class JobSubmissionSlownessRegressionTest extends JetTestSupport {
         DAG dag = new DAG();
         Vertex v1 = dag.newVertex("v", noopP());
         Vertex v2 = dag.newVertex("v2", noopP());
-        dag.edge(Edge.between(v1, v2));
+        dag.edge(Edge.between(v1, v2).setConfig(new EdgeConfig().setQueueSize(1)));
         return dag;
     }
 

